tests/test_lightning_configs.py

```diff
@@ -0,0 +1,156 @@
+        },
+        task_model_kwargs={
+            "learning_rate": 5e-4,
+            "train_batch_size": 32,
+            "eval_batch_size": 32,
+            "use_scheduler": False,
+            "optimizer": "AdamW",
+            "adam_epsilon": 1e-8,
+            "warmup_steps": 100,
+            "weight_decay": 0.0,
+        },
+        datamodule_kwargs={
+            "max_seq_length": 64,
+        },
+        early_stopping_kwargs={
+            "monitor": "val/Loss",
+            "mode": "min",
+            "patience": 3,
+        },
+    )
+
+
+@pytest.fixture(scope="module")
+def lightning_config_missing_parameters() -> LightningAdvancedConfig:
+    return LightningAdvancedConfig(
+        finetune_last_n_layers=0,
+        task_train_kwargs={
+            "max_epochs": 1,
+            "deterministic": True,
+        },
+        task_model_kwargs={
+            "learning_rate": 5e-4,
+            "train_batch_size": 32,
+            "eval_batch_size": 32,
+            "use_scheduler": False,
+        },
+        datamodule_kwargs={
+            "max_seq_length": 64,
+        },
+        early_stopping_kwargs={
+            "monitor": "val/Loss",
+            "mode": "min",
+            "patience": 3,
+        },
+    )
+
+
+def test_lightning_config_missing_params(
+    lightning_config_missing_parameters: LightningConfig,
+    dataset_kwargs: Dict[str, Any],
+    result_path: "TemporaryDirectory[str]",
+) -> None:
+    pipeline = LightningClassificationPipeline(
+        embedding_name_or_path="allegro/herbert-base-cased",
+        output_path=result_path.name,
+        config=lightning_config_missing_parameters,
+        devices="auto",
+        accelerator="cpu",
+        **dataset_kwargs[0],
+    )
+    with pytest.raises(TypeError):
+        pipeline.run()
+
+
+def test_lightning_advanced_config_from_basic() -> None:
+    basic_config = LightningBasicConfig()
+    config = LightningAdvancedConfig.from_basic()
+    for attr in config.__annotations__:
+        assert getattr(config, attr) == getattr(basic_config, attr)
+
+
+def test_lightning_classification_basic_config(
+    lightning_basic_config: LightningConfig,
+    dataset_kwargs: Dict[str, Any],
+    result_path: "TemporaryDirectory[str]",
+) -> None:
+    pl.seed_everything(441, workers=True)
+    pipeline = LightningClassificationPipeline(
+        embedding_name_or_path="allegro/herbert-base-cased",
+        output_path=result_path.name,
+        config=lightning_basic_config,
+        devices="auto",
+        accelerator="cpu",
+        **dataset_kwargs[0],
+    )
+    pipeline.run()
+
+
+def test_lightning_classification_advanced_config(
+    lightning_advanced_config: LightningConfig,
+    dataset_kwargs: Dict[str, Any],
+    result_path: "TemporaryDirectory[str]",
+) -> None:
+    pl.seed_everything(441, workers=True)
+    pipeline = LightningClassificationPipeline(
+        embedding_name_or_path="allegro/herbert-base-cased",
+        output_path=result_path.name,
```

Comment:
if you do not check predictions, you can use random model ""hf-internal-testing/tiny-albert" to speed up tests.

Reply:
changed